### ISSUE TITLE
What is a "proprietary instance" as mentioned on the README?

### ISSUE THREAD

**Issue by 12joan** [zedeus/nitter#922]
I noticed that Nitter's README states that "no proprietary instances [are] permitted" in connection with Nitter's AGPLv3 license. I was wondering what the intent of this phrasing was, and what the legal ramifications are?

I'm not a lawyer (and, let's be honest, you need to be a lawyer to have any chance of understanding GPL-family licenses), but [GitHub's info card](https://github.com/zedeus/nitter/blob/master/LICENSE) for AGPLv3 states that both commercial and private use are allowed, so long as the usual conditions are followed.

The only definition of "proprietary instance" I can think of that would be consistant with these permissions is "an instance containing closed-source modifications," in which case the statement that such instances are not permitted would be an effort to highlight this information for those who aren't familiar with AGPLv3.

The other possibility I can think of is that "proprietary instance" was intended to have some definition not already covered by (or contrary to) AGPLv3, such as "instances to whom only a select group of people are granted access, possibly in exchange for payment." Such instances, while not prohibited by AGPLv3, may be out of line with Nitter's ethos, in which case prohibiting them explicitly may be the desired effect of this line.

I won't speculate on whether the phrase "no proprietary instances permitted" has the power to contradict the permissions already granted in the project's LICENSE file, but I'll happily respect the team's wishes if this was the intended meaning.

**Comment by HookedBehemoth** [comment]
The AGPL compared to base GPL includes usage over the network. If you offer nitter somewhere, your users should be able to request access.
Commercial use has nothing to do with this. You can charge people for using your nitter instance but you still have to share the source code modifications.

**Comment by 12joan** [comment]
Got it, so "proprietary" here is referring to closed-source modifications. Thanks for clarifying! 🙂

Perhaps this could be rephrased on the README to avoid ambiguity? Perhaps something like, "Any changes you make to Nitter's source code must be made available to your users."

**Comment by markcellus** [comment]
Yeah I think your first thought is correct. Loved how well you wrote this issue btw 😍.

**Comment by OIRNOIR** [comment]
Just to clarify, closed-source modifications are OK if the instance can and will only be used by you? (AKA a private instance)

**Comment by gldrk** [comment]
> Just to clarify, closed-source modifications are OK if the instance can and will only be used by you? (AKA a private instance)

The notion of ‘closed source’ is only applicable to works exposed to non-copyright holders.  Private modifications are trivially free and no free software license is allowed to impose conditions on them.  See https://www.gnu.org/philosophy/free-sw.html.
>You should also have the freedom to make modifications and use them privately in your own work or play, without even mentioning that they exist.

**Comment by 12joan** [comment]
> The notion of ‘closed source’ is only applicable to works exposed to **non-copyright holders**

This interpretation could have some interesting edge cases when taken to extremes. For instance, what if I dedicate copyright to my changes to the public domain but refuse access to the content of those changes (i.e. the source code)?

I've no doubt you're correct that modifications used privately don't trigger [A]GPL's conditions, but the process of determining what counts as private can be...[complex](https://opensource.stackexchange.com/a/12858). That's only if your idea of "private" includes you and everyone in your company, though. I'm pretty sure it's fine if it really is just you, as in the case @OIRNOIR describes.

**Comment by untitaker** [comment]
See also https://github.com/zedeus/nitter/issues/982
